akka-typed-tests/src/test/java/akka/typed/ExtensionsTest.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed;
+
+import akka.actor.*;
+import com.typesafe.config.ConfigFactory;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.Optional;
+
+import static junit.framework.TestCase.assertSame;
+import static org.junit.Assert.assertTrue;
+
+public class ExtensionsTest extends JUnitSuite {
+
+
+  public static class MyExtImpl implements Extension {
+  }
+
+  public static class MyExtension extends ExtensionId<MyExtImpl> {
+
+    private final static MyExtension instance = new MyExtension();
+
+    private MyExtension() {
+    }
+
+    public static MyExtension getInstance() {
+      return instance;
+    }
```

Comment:
Skipping the `ExtensionIdProvider` boilerplate and just say that in Scala - use `object` for extensions that can be loaded through config, and for Java - provide a singleton accessor which we call using reflection (and we are anyways doing reflections trixery in the loading of extensions from config).